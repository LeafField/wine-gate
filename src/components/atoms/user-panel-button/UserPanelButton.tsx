import React, { FC } from "react";
import Link from "next/link";

type Props = {
  href: string;
  title: string;
};

const UserPanelButton: FC<Props> = ({ href, title }) => {
  return (
    <Link
      className="user-panel-button hover:border-blue focus:outline-blue"
      href={href}
    >
      <span className="w-24 text-center">{title}</span>
    </Link>
  );
};

export default UserPanelButton;
