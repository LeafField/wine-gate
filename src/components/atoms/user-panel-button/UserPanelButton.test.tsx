import { screen, render } from "@testing-library/react";
import UserPanelButton from "./UserPanelButton";

describe("UserPanelButtonの単体テスト", () => {
  test("UserPanelButtonが表示されているか", () => {
    render(<UserPanelButton href="/" title="ワイン記事を投稿する" />);
    expect(screen.getByText("ワイン記事を投稿する")).toBeInTheDocument();
  });

  test("スナップショットテスト", () => {
    const { container } = render(
      <UserPanelButton href="/" title="ワイン記事を投稿する" />,
    );
    expect(container).toMatchSnapshot();
  });
});
