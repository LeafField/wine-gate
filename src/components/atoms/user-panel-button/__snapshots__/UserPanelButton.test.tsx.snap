// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPanelButtonの単体テスト スナップショットテスト 1`] = `
<div>
  <a
    class="user-panel-button hover:border-blue focus:outline-blue"
    href="/"
  >
    <span
      class="w-24 text-center"
    >
      ワイン記事を投稿する
    </span>
  </a>
</div>
`;
